project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/model/item/index/CompositeIndex.kt

```diff
@@ -0,0 +1,30 @@
+package org.plan.research.minimization.plugin.model.item.index
+
+import org.plan.research.minimization.plugin.psi.stub.KtStub
+
+import arrow.core.raise.option
+import com.intellij.psi.PsiElement
+import fleet.util.indexOfOrNull
+import org.jetbrains.kotlin.psi.KtClass
+import org.jetbrains.kotlin.psi.KtNamedFunction
+
+import kotlin.collections.filter
+
+sealed class CompositeIndex : PsiChildrenPathIndex {
+    data class ChildrenNonDeclarationIndex(val childrenIndex: Int) : CompositeIndex() {
+        override fun getNext(element: PsiElement): PsiElement? =
+            element.children.filter { it !is KtClass && it !is KtNamedFunction }.getOrNull(childrenIndex)
```

Comment:
I see two places with the same code, maybe make a function for it?
And is there any common class/interface for instructions/non-declaration psi elements?

Reply:
I thought copy-paste in the small class in okay bc the only reason for its existence is to have it :)